cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -53,16 +53,16 @@ type clientCertAuth struct {
 
 // tokenAuth struct holds info required to use a token to provide authentication info in a kubeconfig object
 type tokenAuth struct {
-	Token string
+	Token string `datapolicy:"token"`
 }
 
 // kubeConfigSpec struct holds info required to build a KubeConfig object
 type kubeConfigSpec struct {
 	CACert         *x509.Certificate
 	APIServer      string
 	ClientName     string
-	TokenAuth      *tokenAuth
-	ClientCertAuth *clientCertAuth
+	TokenAuth      *tokenAuth `datapolicy:"token"`
+	ClientCertAuth *clientCertAuth `datapolicy:"security-key"`
```

Comment:
the KEP mentions
> Recursion will stop on pointers as values for those are usually not logged

wouldn't that mean that TokenAuth and ClientCertAuth would not be filtered regardless as pointer values in the struct?
also kubeConfigSpec or it's fields are not logged during runtime.

Reply:
> Recursion will stop on pointers as values for those are usually not logged

I think (or hope) this is a typo or misphrasing in the KEP?  While (raw) pointers are rarely meaningful in logs, I feel like pointers are routinely logged on the understanding that they will render in logs as `&{foo bar baz}`.

As a result, any log call with a tagged reference value, like those here, _should_ be filtered, because the underlying value will be rendered otherwise.  However, for pointers of greater depth, e.g. `TokenAuthRef **tokenAuth`, recursion could be halted, since logging an address is unlikely to present a security risk.

> also kubeConfigSpec or its fields are not logged during runtime.

And indeed it should not be, exactly because of the potential of leaking these credentials.  KEP-1753 and KEP-1933 are defensive measures against future developer mistakes.  Such mistakes [are not uncommon](https://groups.google.com/g/kubernetes-security-announce/c/9d0gPe7SCM8).

Reply:
@PurelyApplied Thoughs?

Reply:
looks like the KEP might need a minor clarification around the referenced sentence.

Reply:
I think `klog.Infof("%+v", foo{pointer: &bar{baz: "secret"}})` won't print `secret`, but `klog.Infof("%v", &bar{baz: "secret"})` will, so only on the initial argument do we need to dereference pointers, not while recursing, is my understanding correct?